### ISSUE TITLE
Space between method name and parenthesis not detected

### ISSUE THREAD

**Issue by Mayeu** [rubocop/rubocop#435]
Hi,
In the ruby style guide:

```
Never put a space between a method name and the opening parenthesis.
```

If I create the following code:

``` ruby
def method (var)
    var
end
```

Rubocop does not return an offence for the space between the method name and the parenthesis.

Neither does it in this case:

``` ruby
result = method_called (with_var)
```

**Comment by jaredbeck** [comment]
> It looks like this should be implemented in two separate cops because running ruby -wc on
> 
>    result = method_called (with_var)
>
> gives this warning:
> 
>    warning: (...) interpreted as grouped expression

This is no longer true in ruby >= 2.1, was true in ruby 1.9.3. I'm not sure why ruby decided to drop this warning. I'm not sure exactly when it changed either, maybe [ruby bug 5214](https://bugs.ruby-lang.org/issues/5214)?

**Comment by jonas054** [comment]
@jaredbeck Removing that warning doesn't make much sense to me, and I think RuboCop is doing the right thing by reporting, especially if the example is a little more complex, like
```ruby
def method_called(x)
  x * 2
end

result = method_called (1) + 2
p result
```
This program prints `6` since the method call line is parsed as `result = method_called((1) + 2)`, but it's certainly possible that the user meant `result = method_called(1) + 2`.

**Comment by jaredbeck** [comment]
```
method_called (1) + 2
```

Aha! Thanks Jonas. This is a great example. We should add this one to the docs at the top of `parentheses_as_grouped_expression.rb`
